test/tests/services/resources-getter.test.js

```diff
@@ -273,4 +310,44 @@ describe('service > resources-getter', () => {
       });
     });
   });
+
+  describe('projection feature', () => {
+    describe('with selected smartfield', () => {
+      it('should return all fields', async () => {
+        expect.assertions(3);
+        const parameters = {
+          fields: { films: 'description' },
+          page: { number: '1', size: '15' },
+          searchExtended: '0',
+          timezone: '+01:00',
+        };
+
+        const result = await new ResourcesGetter(FilmModel, options, parameters).perform();
+        expect(result[0]).toHaveLength(3);
+        const titles = result[0].filter((film) => !!film.title);
+        expect(titles).toHaveLength(3);
+        const durations = result[0].filter((film) => !!film.duration);
+        expect(durations).toHaveLength(3);
```

Comment:
Opinion mode, sorry, please ignore me! (I removed the `[0]` because ... uh ... I feel it noisy and I added a few new lines)

```javascript
const [result] = await new ResourcesGetter(FilmModel, options, parameters).perform();
expect(result).toHaveLength(3);

const titles = result.filter((film) => !!film.title);
expect(titles).toHaveLength(3);

const durations = result.filter((film) => !!film.duration);
expect(durations).toHaveLength(3);
```